In [1]:
import pandas as pd
import os

In [2]:
r_23= pd.ExcelFile("data_downloads_website/2022-2023_england_ks5underlying.xlsx")

In [3]:
r_23.sheet_names

['Contents',
 'Metadata',
 'Institution_subject_results',
 'Institution_aggregated_results',
 'National_data',
 'Institution_subject_entries']

In [4]:
subj_23 = pd.read_excel(r_23, "Institution_subject_results")  

In [5]:
subj_23.columns

Index(['Year', 'Local Authority', 'URN', 'School or college name',
       'School or college type', 'Exam cohort', 'Qualification', 'Level',
       'ASIZE', 'GSIZE', 'Grade structure', 'Subject', 'Grade/Total entries',
       'Number of exams'],
      dtype='object')

In [6]:
subj_23.shape

(582314, 14)

In [ ]:
schools_urn = [136298,138313,137353,100750,132942]

In [14]:
m = subj_23.loc[subj_23['URN']==137353].copy()

In [15]:
m.shape

(176, 14)

Year
2023    176
Name: count, dtype: int64
Local Authority
Westminster    176
Name: count, dtype: int64
URN
137353    176
Name: count, dtype: int64
School or college name
The St Marylebone CofE School    176
Name: count, dtype: int64
School or college type
Converter Academy    176
Name: count, dtype: int64
Exam cohort
A level           168
Other academic      8
Name: count, dtype: int64
Qualification
GCE A level                   162
Extended Project (Diploma)      8
GCE AS level                    6
Name: count, dtype: int64
Level
3    176
Name: count, dtype: int64
ASIZE
1.0    162
0.3      8
0.5      6
Name: count, dtype: int64
GSIZE
4.0    162
1.2      8
2.0      6
Name: count, dtype: int64
Grade structure
*,A,B,C,D,E    170
A,B,C,D,E        6
Subject

(162, 14)

In [19]:
m[['Qualification', 'Level',
       'ASIZE', 'GSIZE', 'Grade structure', 'Subject', 'Grade/Total entries',
       'Number of exams']].head()

Qualification  Level  ASIZE  GSIZE Grade structure  \
386133   GCE A level      3    1.0    4.0     *,A,B,C,D,E   
386134   GCE A level      3    1.0    4.0     *,A,B,C,D,E   
386135   GCE A level      3    1.0    4.0     *,A,B,C,D,E   
386136   GCE A level      3    1.0    4.0     *,A,B,C,D,E   
386137   GCE A level      3    1.0    4.0     *,A,B,C,D,E   

                          Subject Grade/Total entries Number of exams  
386133  Art and Design (Fine Art)                   *              12  
386134  Art and Design (Fine Art)                   A              19  
386135  Art and Design (Fine Art)                   B               2  
386136  Art and Design (Fine Art)                   C               0  
386137  Art and Design (Fine Art)                   D               0

In [21]:
m.loc[m['Qualification']=='GCE A level'].pivot(index='Subject', columns='Grade/Total entries', values='Number of exams')

Grade/Total entries           *     A     B     C     D     E Fail Total
Subject                                                                 
Art and Design (Fine Art)    12    19     2     0     0     0  NaN    33
Biology                       4     7     6     5     5     1  NaN    28
Chemistry                     1     7     5     8     6     1  NaN    28
Classical Civilisation        1     1     4     5     0     0  NaN    11
Dance                         3     2     1     0     0     0  NaN     6
Drama and Theatre Studies     5     9     8     0     0     0  NaN    22
Economics                     1     8    13    11     4     0  NaN    37
English Literature           13    13    13     1     1     0  NaN    41
French                        3     2     3     4     0     1  NaN    13
Geography                     3     7    11     4     2     0  NaN    27
German                     Supp  Supp  Supp  Supp  Supp  Supp  NaN     1
Government and Politics       6     3     4     1     0     0  NaN    14
History                       7    13    10     4     1     0  NaN    35
Mathematics                  12     9     8     7     4     1  NaN    41
Mathematics (Further)         2     4     1     1     1     1  NaN    10
Media/Film/Tv Studies      Supp  Supp  Supp  Supp  Supp  Supp  NaN     5
Music                         3     3     1     0     0     0  NaN     7
Music Technology              1     2     3     2     0     1  NaN     9
Physics                       1     5     1     1     2     1    1    12
Psychology                    7    13    14     9     3     1  NaN    47
Religious Studies             2     3     3     2     0     0  NaN    10
Sociology                     1     7     8     4     0     0  NaN    20
Spanish                       2     1     2     3     2     0  NaN    10